python/cudf/cudf/tests/test_dataframe.py

```diff
@@ -7190,75 +7213,46 @@ def test_dataframe_sample_basic(n, frac, replace, axis):
                     "n": n,
                     "frac": frac,
                     "replace": replace,
-                    "random_state": random_state,
+                    "random_state": gd_random_state,
+                    "weights": gd_weights,
                     "axis": axis,
                 },
             ),
+            compare_error_message=False,
         )
     else:
-        gout = df.sample(
+        got = df.sample(
             n=n,
             frac=frac,
             replace=replace,
-            random_state=random_state,
+            random_state=gd_random_state,
+            weights=gd_weights,
             axis=axis,
         )
-        assert pout.shape == gout.shape
+        checker(expected, got)
 
 
 @pytest.mark.parametrize("replace", [True, False])
-@pytest.mark.parametrize("random_state", [1, np.random.mtrand.RandomState(10)])
-def test_dataframe_reproducibility(replace, random_state):
+def test_dataframe_sample_reproducibility(replace, random_state_tuple):
     df = cudf.DataFrame({"a": cupy.arange(0, 1024)})
 
-    expected = df.sample(1024, replace=replace, random_state=random_state)
-    out = df.sample(1024, replace=replace, random_state=random_state)
+    _, gd_random_state, _ = random_state_tuple
```

Comment:
This test shouldn't be parameterized by the `random_state_tuple`. Just parametrize directly by the random state lib.

If `gd_random_state` is an int, this is going to end up constructing the random state as `type(42)(10)` which is both unnecessary and a super roundabout way of passing in a seed value. I'd just skip that entirely, or if you really want you can put an if statement inside the test that constructs a random state based on whether the input is an int or a random state object.